agent/structs/connect_ca.go

```diff
@@ -380,6 +382,17 @@ func (c CommonCAProviderConfig) Validate() error {
 		return nil
 	}
 
+	// if the root cert ttl is not set, set it to the default
+	if c.RootCertTTL == 0 {
+		c.RootCertTTL = 10 * 365 * 24 * time.Hour
```

Comment:
I think we can do `c.RootCertTTL = time.ParseDuration(DefaultRootCertTTL)` here and use the default that's already defined above

Reply:
This function is `Validate`, but here we are normalizing (not validating).

I suspect this should this return an error instead?  In normal operations we expect `RuntimeConfig` to always have set a default, so users should only encounter this problem if they explicit set a 0 for TTL (I think).

Reply:
> but here we are normalizing 
>  In normal operations we expect RuntimeConfig to always have set a default, 

Hey both, thanks for the engagement here -- I think per Daniel's comment, I'm inclined to take out this chunk of code.

If a callee actually sets the value to less than the intermediate cert ttl than the right error should come out.

I also see no such "normalization"/ defaulting for leaf and intermediate ttls